### ISSUE TITLE
[AMD + Screamer M.2 R03] pcileech probe reads 10% then fails remaining

### ISSUE THREAD

**Issue by dedbeef** [ufrisk/pcileech#149]
I've tested the card extensively on an Intel Z170 board and have no issues, it runs very reliably.
Testing on an AMD B450 (Aorus) using the command `.\pcileech probe -device fpga -v -vv`

I have disabled CSM, Secure Boot, IOMMU, SVM Mode and SMT while testing (is there anything else I'm missing?).
The probe works fine for the first 10% of pages, then fails the remaining pages.
Then when running the command again, it fails 100%.

Card is detecting fine it seems and LD1 and LD3 are lit indicating a solid connection.
`DEVICE: FPGA: ScreamerM2 PCIe gen2 x1 [300,0,500] [v4.3,0700]`

I've also tried in the M.2 port, in a PCI-e port via the included adapter and via another PCI-e port to be on the safe side.
Hoping you can point out something I can try, thanks in advance.

**First probe**
```
LcMemMap_AddRange: 0000000000000000-000000000009ffff -> 0000000000000000
LcMemMap_AddRange: 0000000000100000-000000081f37ffff -> 0000000000100000
LeechCore v2.0.2: Open Device: fpga
 Memory Map:
 START              END               #PAGES
 0000000000000000 - 000000000009ffff  000000a0
 00000000000c0000 - 00000000deffffff  000def40

 Current Action: Probing Memory
 Access Mode:    Normal
 Progress:       33267 / 33267 (100%)
 Speed:          386 MB/s
 Address:        0x000000081F380000
 Pages read:     913376 / 8516480 (10%)
 Pages failed:   7603104 (89%)
Memory Probe: Completed.
```

**Subsequent Probes**
```
FPGA: TINY PCIe TLP algrithm auto-selected!
LcMemMap_AddRange: 0000000000000000-000000000009ffff -> 0000000000000000
LcMemMap_AddRange: 0000000000100000-00000000ffffffff -> 0000000000100000
LeechCore v2.0.2: Open Device: fpga
 Memory Map:
 START              END               #PAGES


 Current Action: Probing Memory
 Access Mode:    Normal
 Progress:       4096 / 4096 (100%)
 Speed:          341 MB/s
 Address:        0x0000000100000000
 Pages read:     0 / 1048576 (0%)
 Pages failed:   1048576 (100%)
Memory Probe: Completed.
```

**Comment by dedbeef** [comment]
I was gobsmacked when I found out you are self-funded (small mention in an issue) and will be talking to my colleagues to garner some more support for your extremely useful work.

And much appreciated, that makes perfect sense. I'm currently going through a the debugging process with 2 other new R04s and we may have stumbled on some other things that improved the stability dramatically. I'll post more info as soon as we've tested these ideas empirically.
